api/v1/notification_types.go

```diff
@@ -0,0 +1,63 @@
+package v1
+
+import (
+	"github.com/flanksource/incident-commander/api"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:object:generate=true
+type NotificationSpec struct {
+	// List of events that can trigger this notification
+	Events []string `json:"events" yaml:"events"`
+
+	// The title for the notification
+	Title string `json:"title,omitempty" yaml:"title,omitempty"`
+
+	// Template is the notification body in markdown
+	Template string `json:"template,omitempty" yaml:"template,omitempty"`
+
+	// Cel-expression used to decide whether this notification client should send the notification
+	Filter string `json:"filter,omitempty" yaml:"filter,omitempty"`
+
+	// email or ID of the recipient person
+	Person string `json:"person,omitempty" yaml:"person,omitempty"`
+
+	// name or ID of the recipient team
+	Team string `json:"team,omitempty" yaml:"team,omitempty"`
+
+	// Custom services to send this notification to
+	CustomServices []api.NotificationConfig `json:"custom_services,omitempty" yaml:"custom_services,omitempty"`
+
+	// Properties for Shoutrrr
+	Properties map[string]string `json:"properties,omitempty" yaml:"properties,omitempty"`
```

Comment:
I think the spec should look something like this:
```yaml
to:
   person: 
   team: 
   connection:
   url: 
   properties:
```

I don't think we need to support multiple custom services